coffeecard/CoffeeCard.Library/Services/TokenService.cs

```diff
@@ -44,28 +44,65 @@ public JwtSecurityToken ReadToken(string token)
             return new JwtSecurityTokenHandler().ReadToken(token) as JwtSecurityToken;
         }
 
+        
         /// <summary>
-        ///     Receives the serialized version of a JWT token as a string. Checks if the JWT token is valid (Based on its lifetime
-        ///     and if it has been used before)
+        /// Receives the serialized version of a JWT token as a string.
+        /// Checks if the JWT token is valid (Based on its lifetime)
         /// </summary>
         /// <param name="tokenString"></param>
-        /// <returns></returns>
-        public async Task<bool> ValidateToken(string tokenString)
+        /// <returns>bool</returns>
+        public async Task<bool> ValidateTokenAsync(string tokenString)
+        {
+            var securityKey = new SymmetricSecurityKey(
+                Encoding.UTF8.GetBytes(_identitySettings.TokenKey));
+
+            try
+            {
+                var securityTokenHandler = new JwtSecurityTokenHandler();
+                var validationParameters = new TokenValidationParameters
+                {
+                    ValidateAudience = false,
+                    ValidateIssuer = false,
+                    ValidateIssuerSigningKey = true,
+                    IssuerSigningKey = securityKey,
+                    ValidateLifetime = true,
+                    ClockSkew = TimeSpan.Zero //the default for this setting is 5 minutes
+                };
+
+                securityTokenHandler.ValidateToken(tokenString, validationParameters, out _); // Throws exception if token is invalid
+            }
+            catch (Exception e) when (e is ArgumentException ||
+                                      e is SecurityTokenException)
+            {
+                Log.Information("Received invalid token");
```

Comment:
Do you think it would make sense to log more, e.g. the stack trace, and the token (maybe in a debug log statement) just to have some logging and debug hints if this start failing?

Reply:
The token itself could make sense to log in a debug statement, but I find it very unlikely that we will be looking at it. I find it unlikely that this check will start failing, since we are making use of the same validation parameters here, that we use on the API as a whole.